### PR TITLE
Mapreduce code

### PR DESCRIPTION
**Pull request by zhouhanc**
@pelmers , @jfking 

Can you guys review it?

### CODE REVIEW

File: rice-namenode/include/ClientNamenodeProtocolImpl.h [hunk 1]
```diff
@@ -51,6 +52,10 @@ class ClientNamenodeTranslator {
 		std::string setPermission(std::string);
 		std::string recoverLease(std::string);
 		std::string concat(std::string);
+		std::string getEZForPath(std::string input);
+		std::string setOwner(std::string input);
+		std::string getContentSummary(std::string input);
```
Comment (pelmers) [inline rice-namenode/include/ClientNamenodeProtocolImpl.h]: looks like we support this one, let's move it up in the header so it's not under the section "not supported"

File: vagrant_provision.sh [hunk 2]
```diff
@@ -53,20 +54,55 @@ cat > /home/vagrant/hadoop/etc/hdfs-site.xml <<EOF
...
+datadir=/var/lib/mysql
+socket=/var/lib/mysql/mysql.sock
+bind-address=127.0.0.1
+default-storage-engine=InnoDB
+sql_mode=STRICT_ALL_TABLES
+EOF
+service mysql start
+sysv-rc-conf mysql on
+
+# TODO: finish this Setup Apache Hue and tell it to use mysql
```
Comment (pelmers) [inline vagrant_provision.sh]: I self-assign this. @pelmers

File: zookeeper/source/zk_nn_client.cc [hunk 3]
```diff
@@ -384,7 +384,8 @@ namespace zkclient{
 		}
 		if (file_blocks.size() == 0) {
 			LOG(ERROR) << "No blocks found for file " << ZookeeperPath(src);
-			res.set_result(false);
+			//res.set_result(false);
+			res.set_result(true);
```
Comment (pelmers) [inline zookeeper/source/zk_nn_client.cc:388]: why do we set true here? is that what hive expects?

Comment (zhouhanc) [reply zookeeper/source/zk_nn_client.cc:388]: I found in mapreduce tasks, sometimes a file is created but the size of the file is 0, but we still want to return true to the client.

File: test/namenode/NameNodeTest.cc [hunk 4]
```diff
@@ -199,42 +199,122 @@ namespace {
...
+		ASSERT_EQ(true, client->previousBlockComplete(block_id));
 	}
 
 	TEST_F(NamenodeTest, testRenameFile){
 		int error_code;
-		zk->create("/fileSystem/old_name", zk->get_byte_vector("File data"), error_code, false);
-		ASSERT_EQ(0, error_code);
 
+		// Create a test file for renaming
+        hadoop::hdfs::CreateRequestProto create_req;
```
Comment (jfking) [inline test/namenode/NameNodeTest.cc]: Minor, but can you fix the spacing on these two lines? Should be tabs

File: test/namenode/NameNodeTest.cc [hunk 5]
```diff
@@ -199,42 +199,122 @@ namespace {
...
 
-		ASSERT_TRUE(client->rename_file("/old_name", "/new_name"));
-
-		auto new_file_data = std::vector<std::uint8_t>();
-		zk->get("/fileSystem/new_name", new_file_data, error_code);
-		ASSERT_EQ(0, error_code);
-		ASSERT_EQ("File data", std::string(new_file_data.begin(), new_file_data.end()));
-		LOG(INFO) << "New: " << std::string(new_file_data.begin(), new_file_data.end());
-
+		// Rename
```
Comment (jfking) [inline test/namenode/NameNodeTest.cc:228]: Also down here for the spacing. Just find-replace all instances of 4 spaces in the file to a tab character :)

**Comment by jfking** [comment]
Looks good overall. I'm not very familiar with what's going on, but I couldn't find anything obviously wrong and the tests look good.

**Comment by zhouhanc** [comment]
Merged!

**Pull request merged**